main.go

```diff
@@ -34,12 +28,27 @@ func main() {
 		log.Fatal("Error: ", err)
 	}
 
-	var timeoutChan <-chan time.Time
+	ctx := context.Background()
+	var cancel func()
 	if args.Timeout != time.Duration(0) {
-		timeoutChan = time.After(args.Timeout)
+		ctx, cancel = context.WithTimeout(ctx, args.Timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
 	}
+	defer cancel()
 
-	jobIn, err := parseAndValidateJob(args.JobFile)
+	go func() {
+		sigChan := make(chan os.Signal)
+		signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+		// wait for signal
+		<-sigChan
+
+		// cancel context
+		cancel()
```

Comment:
Is that the same cancel as in line 38?
What happens if `defer` tries to cancel that context again?

Reply:
https://golang.org/pkg/context/#CancelFunc
"After the first call, subsequent calls to a CancelFunc do nothing."